### DISCUSSION TITLE
Trigger Charge/ Discharge based on changing energy prices

### DISCUSSION THREAD

**Discussion by cchurch132** [hif2k1/battery_sim#84] [General]
Is it possible to trigger force charge/ discharge based on a varying energy price? I'm guessing it would need to have a real-time feed of the price to do so?

**Comment by hif2k1** [comment]
If you have a sensor that gives you a live reading of the energy price then
you can create an automation in the normal way and get it to switch the
battery into forced charging mode yes.

**Reply by cchurch132** [reply]
Thanks. Is it possible to force discharge to simulate feeding into the grid and also use force charge to charge from the grid? Obviously not at the same time, two different situations depending on the buy or sell price at the time.

**Reply by hif2k1** [reply]
Hopefully this is now implemented and working.
